src/server/game/Entities/Player/Player.cpp

```diff
@@ -11370,6 +11329,33 @@ InventoryResult Player::CanUseItem(ItemTemplate const* proto) const
     if ((proto->GetAllowableClass() & getClassMask()) == 0 || (proto->GetAllowableRace() & getRaceMask()) == 0)
         return EQUIP_ERR_CANT_EQUIP_EVER;
 
+    if (proto->GetSkill() != 0)
+    {
+        bool allowEquip = false;
+        uint32 itemSkill = proto->GetSkill();
+        // Armor that is binded to account can "morph" from plate to mail, etc. if skill is not learned yet.
+        if (proto->GetQuality() == ITEM_QUALITY_HEIRLOOM && proto->GetClass() == ITEM_CLASS_ARMOR && !HasSkill(itemSkill))
+        {
+            /// @todo when you right-click already equipped item it throws EQUIP_ERR_PROFICIENCY_NEEDED.
```

Comment:
this one here sounds suspicious. this usually implies mismatching client-server logic

Reply:
Not sure, I didn't add/change any of this logic. Just moved it from `Player::CanUseItem(Item* pItem, bool not_loading)` above